packages/data-table-server/src/dataTableService/services/DataGroupMetaDataDataTableService.ts

```diff
@@ -0,0 +1,84 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2022 Beyond Essential Systems Pty Ltd
+ */
+
+import { AccessPolicy } from '@tupaia/access-policy';
+import { TupaiaApiClient } from '@tupaia/api-client';
+import { Aggregator } from '@tupaia/aggregator';
+import { DataBroker } from '@tupaia/data-broker';
+import { upperFirst, yup } from '@tupaia/utils';
+import { DataTableService } from '../DataTableService';
+
+const requiredParamsSchema = yup.object().shape({
+  dataGroupCode: yup.string().required(),
+});
+
+const configSchema = yup.object();
+
+type DataGroupMetaDataDataTableServiceContext = {
+  apiClient: TupaiaApiClient;
+  accessPolicy: AccessPolicy;
+};
+type DataGroup = {
+  code: string;
+  name: string;
+  dataElements: { code: string; name: string; text: string; options: Record<string, unknown>[] }[];
+};
+
+type DataGroupMetadata = {
+  dataGroupCode: string;
+  dataGroupName: string;
+  dataElementCode: string;
+  dataElementName: string;
+  [key: string]: unknown;
+};
```

Comment:
Do a bit data massage here to merge data group metadata with data element metadata, I find it handy to return an array instead of an object for viz-builder.